health/default.go

```diff
@@ -0,0 +1,75 @@
+package health
+
+import (
+	"sync"
+
+	"google.golang.org/grpc"
+)
+
+var (
+	// DefaultServer is the Server instance on which the package-level functions
+	// operate. Most servers need only a single health server so this provides
+	// a convenient definition of it that is available everywhere.
+	DefaultServer *Server
+	defaultState  State
+
+	serverInit sync.Once
+)
+
+// RegisterWithGRPC registers the default server
+// health.DefaultServer.GRPC with the given grpc Server to make the
+// health service available at "anz.health.v1.Health". This
+// RegisterWithGRPC function returns an error when the Version
+// information is invalid.
+func RegisterWithGRPC(s *grpc.Server) error {
+	var err error
+	serverInit.Do(func() { err = newDefaultServer() })
+	if err != nil {
+		return err
+	}
+	DefaultServer.GRPC.RegisterWith(s)
+	return nil
+}
+
+// RegisterWith registers the default server health.DefaultServer.HTTP
+// with the given Router, e.g. a http.ServeMux, to make the health
+// service endpoints available at /healthz, /readyz and /version. This
+// RegisterWith function returns an error when the Version information
+// is invalid.
+func RegisterWith(r Router) error {
```

Comment:
I'm not sure why you've broken the symmetry with `RegisterWithGRPC`. Shouldn't this be `RegisterWithHTTP` ?

Reply:
fwiw, I did think about it and consciously decided against it, because it seems HTTP is the more common use case so I wanted to make it easier.
However, I acknowledge your argument that in that case everything should be reworked so let's stick it in.